game-app/game-core/src/main/java/games/strategy/triplea/delegate/battle/steps/fire/general/TargetGroup.java

```diff
@@ -65,15 +64,21 @@ public static List<TargetGroup> newTargetGroups(
   }
 
   private static Set<UnitType> findTargets(
-      final UnitType unitType, final Set<UnitType> unitTypes, final Set<UnitType> enemyUnitTypes) {
-    final Set<UnitType> targets = new HashSet<>(enemyUnitTypes);
-    targets.removeAll(unitType.getUnitAttachment().getCanNotTarget());
-    return unitTypes.stream().anyMatch(Matches.unitTypeIsDestroyer())
-        ? targets
-        : targets.stream()
-            .filter(
-                target -> !target.getUnitAttachment().getCanNotBeTargetedBy().contains(unitType))
-            .collect(Collectors.toSet());
+      UnitType unitType, boolean destroyerPresent, Set<UnitType> enemyUnitTypes) {
+    Set<UnitType> cannotTarget = unitType.getUnitAttachment().getCanNotTarget();
+    // Note: uses a single stream instead of a sequence of removeAll() calls for performance.
```

Comment:
This comment will be confusing to those not looking at this diff!  =D
(Arguably this is something for the commit history)